### PR TITLE
Support mark bounded dynamic and use bound values for shape comparison

### PR DESCRIPTION
**Pull request by yitongh**

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -736,8 +737,12 @@ c10::SymNode XLASymNodeImpl::ne(const c10::SymNode& other) {
 }
 
 c10::SymNode XLASymNodeImpl::gt(const c10::SymNode& other) {
-  XLA_CHECK(false) << "XLASymNodeImpl::" << __FUNCTION__
-                   << " has not been implemented.";
+  TORCH_LAZY_FN_COUNTER_TIMED_TRACING("xla::size_");
+  auto p_other = dynamic_cast<XLASymNodeImpl*>(other.get());
+  XLA_CHECK(is_int()) << __FUNCTION__ << " with non-int NYI";
+  XLA_CHECK(p_other->is_int()) << __FUNCTION__ << " with non-int NYI";
+  auto n_lt = torch::lazy::MakeNode<SizeLt>(p_other->node(), node());
```
Comment (anw90) [inline torch_xla/csrc/tensor.cpp:749]: Add a comment here to explain that SizeLt is used to implement SizeGt.

File: torch_xla/csrc/init_python_bindings.cpp [hunk 2]
```diff
@@ -2329,6 +2329,13 @@ void InitXlaModuleBindings(py::module m) {
           }
           return result;
         });
+  m.def("_xla_mark_bounded_dynamic",
+        [](const at::Tensor& input, const std::vector<uint32_t>& dims,
+           const std::vector<uint32_t>& bounds) {
+          TORCH_LAZY_COUNTER("XlaMarkBoundedDynamic", 1);
+          XLATensorPtr xtensor = bridge::GetXlaTensor(input);
+          xtensor->MarkBoundedDynamicDimension(dims, bounds);
```
Comment (anw90) [inline torch_xla/csrc/init_python_bindings.cpp:2339]: Is there a checking to ensure that len(dims) is equal to len(bounds)?

File: torch_xla/csrc/tensor.cpp (hunk 1 above)
Comment (yitongh) [reply torch_xla/csrc/tensor.cpp:749]: Added

File: torch_xla/csrc/init_python_bindings.cpp (hunk 2 above)
Comment (yitongh) [reply torch_xla/csrc/init_python_bindings.cpp:2339]: I check this in torchacc, also add simple check here.

**Review by anw90 — APPROVED** [review]

**Pull request merged**